api/v1beta1/metal3remediationtemplate_webhook.go

```diff
@@ -42,81 +44,101 @@ var (
 // log is for logging in this package.
 var metal3remediationtemplatelog = logf.Log.WithName("metal3remediationtemplate-resource")
 
-func (r *Metal3RemediationTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
+// Deprecated: This method is going to be removed in a next release.
+func (webhook *Metal3RemediationTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
+		For(webhook).
+		WithDefaulter(webhook, admission.DefaulterRemoveUnknownOrOmitableFields).
+		WithValidator(webhook).
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-metal3remediationtemplate,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=metal3remediationtemplates,versions=v1beta1,name=validation.metal3remediationtemplate.infrastructure.cluster.x-k8s.io,matchPolicy=Equivalent,sideEffects=None,admissionReviewVersions=v1;v1beta1
-// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-metal3remediationtemplate,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=metal3remediationtemplates,versions=v1beta1,name=default.metal3remediationtemplate.infrastructure.cluster.x-k8s.io,matchPolicy=Equivalent,sideEffects=None,admissionReviewVersions=v1;v1beta1
-
-var _ webhook.Defaulter = &Metal3RemediationTemplate{}
-var _ webhook.Validator = &Metal3RemediationTemplate{}
+var _ webhook.CustomDefaulter = &Metal3RemediationTemplate{}
+var _ webhook.CustomValidator = &Metal3RemediationTemplate{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (r *Metal3RemediationTemplate) Default() {
-	if r.Spec.Template.Spec.Strategy.Type == "" {
-		r.Spec.Template.Spec.Strategy.Type = RebootRemediationStrategy
+// Deprecated: This method is going to be removed in a next release.
+func (webhook *Metal3RemediationTemplate) Default(_ context.Context, obj runtime.Object) error {
+	c, ok := obj.(*Metal3RemediationTemplate)
```

Comment:
nit: having a bit of difficulty understanding the reason behind changing the var name from `r` to `c` here

Reply:
no reason, I will change it to m3Rt. I think I changed it trying to satisfy Linter, but in the end issue was somewhere else not here.